ksql-engine/src/main/java/io/confluent/ksql/util/QueryMetadata.java

```diff
@@ -68,8 +79,14 @@ public String getQueryApplicationId() {
   }
 
   public void close() {
-    kafkaStreams.close(100L, TimeUnit.MILLISECONDS);
-    kafkaStreams.cleanUp();
+    kafkaStreams.close();
+    if (kafkaStreams.state() == KafkaStreams.State.NOT_RUNNING) {
```

Comment:
When will this be false?? Is it possible? We should be careful here, because we are definitely leaking resources if this evaluates to false.

Reply:
if wer are using `kafkaStreams#close` with no timeout it is safe to exclude this check. close won't return until the instance has shutdown.
It is only if we wanted to call close with a timeout that we'd need to check the state afterwards